### PR TITLE
Complete remaining task operations

### PR DESCRIPTION
**Pull request by shuzhang1989**

### CODE REVIEW

File: controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/Task.java [hunk 1]
```diff
@@ -27,10 +27,12 @@
  */
 public class Task extends TaskBase {
 
-  public static int PENDING = 0;
-  public static int RUNNING = 1;
-  public static int DONE = 2;
-  public static int FAILED = 3;
+  public enum TaskState {
```
Comment (angxu) [inline controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/Task.java]: There is already a TaskState enum defined in controller-common. You can reuse that one.

File: controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/TaskQueue.java [hunk 2]
```diff
@@ -143,9 +143,10 @@ default boolean removeCluster(final String cluster) {
 
   /**
    * Remove finished tasks.
+   * @param secondsAgo tasks finished seconds ago
    * @return the number of finished tasks which are removed by this call.
    */
-  default int removeFinishedTasks() {
+  default int removeFinishedTasks(final int secondsAgo) {
```
Comment (angxu) [inline controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/TaskQueue.java]: Better to define such API as `#removeFinishedTasks(long, TimeUnit)`.

File: controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/mysql/MySQLTaskQueue.java [hunk 3]
```diff
@@ -138,28 +154,43 @@ public boolean removeCluster(final String clusterName) {
...
-  public boolean enqueueTask(final TaskBase taskBase,
-                             final String clusterName,
-                             final int runDelaySeconds) {
-    entityManager.getTransaction().begin();
+  private long enqueueTaskImpl(final TaskBase taskBase,
+                               final String clusterName,
+                               final int runDelaySeconds,
+                               final Task.TaskState state,
+                               final String claimedWorker,
+                               boolean inTransaction) {
```
Comment (angxu) [inline controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/mysql/MySQLTaskQueue.java]: Or we can just make this method non-transactional and wrap it with `entityManager.getTransaction().begin()` and `entityManager.getTransaction().commit()` in the outer method.

Also for transaction, do we need to do `try {...} finally {...}`?

File: controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/Task.java (hunk 1 above)
Comment (shuzhang1989) [inline controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/Task.java]: sounds good 👍

File: controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/TaskQueue.java (hunk 2 above)
Comment (shuzhang1989) [inline controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/TaskQueue.java]: i think our default time unit is all seconds if you look at other api and design doc.

File: controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/mysql/MySQLTaskQueue.java (hunk 3 above)
Comment (shuzhang1989) [inline controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/mysql/MySQLTaskQueue.java]: sure, how about put transaction().rollback() in finally

**Pull request merged**